source/rust_verify/src/verifier.rs

```diff
@@ -231,6 +236,67 @@ impl Verifier {
         }
     }
 
+    fn print_profile_stats(
+        &self,
+        compiler: &Compiler,
+        profiler: Profiler,
+        qid_map: &HashMap<String, vir::sst::Exp>,
+    ) {
+        let num_quants = profiler.quant_count();
+        let total = profiler.total_instantiations();
+        let max = 10;
+        let delimiter = "-".repeat(100);
+        let msg = format!(
+            "Observed {} total instantiations of user-level quantifiers",
+            total.to_formatted_string(&Locale::en)
+        );
+        compiler.diagnostic().note_without_error(&msg);
+
+        for (index, cost) in profiler.iter().take(max).enumerate() {
+            println!("{}", delimiter);
```

Comment:
Considering you're using diagnostics, can you remove this?

Reply:
That's an interesting idea -- it would certainly be more concise.  

I tried some experiments with the compiler's `Multispan`.  If I create one using `from_spans` using a vector that includes the quantifier's span and the triggers' spans, then we just get a single underline under the whole quantifier (i.e., the quantifier span subsumes the trigger spans).  If I omit the quantifier's span from the vector, it works as long as the quantifier is all on one line, but if it's on multiple lines in the source file, then only the lines that include chosen triggers show up, which seems confusing.

Things work better when I use the quantifier span as the primary span and the triggers as span_labels, or vice versa.  Let me know which of the screenshots below you two like better (if any).  Both are slightly clunky with multi-line quantifiers, but at least they both include all of the relevant info.

![multi1](https://user-images.githubusercontent.com/1816417/173657686-ff79a50d-9c12-4827-ac2b-c52334dc34a1.png)

![multi2](https://user-images.githubusercontent.com/1816417/173657759-d5e93bee-f3d2-4270-93ae-1b7f96e68755.png)

Reply:
i like that the first one has shorter gaps. I don't have a strong preference though

Reply:
@utaal Did you have a preference between the two versions above?

Reply:
(Sorry for the delay!)

I don't have a strong preference; there may be some value in having a more compact output, like @tjhance, but I personally think both would work well.

Thanks a lot for looking into this!

Reply:
Thanks for all of the input!  I pushed the compact version for now.  If we dislike it down the line, we can always tweak it as needed.